frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/BaseTraverseAction.java

```diff
@@ -0,0 +1,129 @@
+package com.virtuslab.gitmachete.frontend.actions.traverse;
+
+import static com.intellij.openapi.application.ModalityState.NON_MODAL;
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getNonHtmlString;
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+
+import java.util.Collections;
+
+import com.intellij.ide.util.PropertiesComponent;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.util.ModalityUiUtil;
+import git4idea.branch.GitBrancher;
+import git4idea.repo.GitRepository;
+import kr.pe.kwonnam.slf4jlambda.LambdaLogger;
+import lombok.CustomLog;
+import lombok.experimental.ExtensionMethod;
+import lombok.val;
+import org.checkerframework.checker.guieffect.qual.UI;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+
+import com.virtuslab.gitmachete.frontend.actions.base.BaseGitMacheteRepositoryReadyAction;
+import com.virtuslab.gitmachete.frontend.actions.base.IBranchNameProvider;
+import com.virtuslab.gitmachete.frontend.actions.dialogs.InfoDialog;
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+import com.virtuslab.gitmachete.frontend.ui.api.table.BaseEnhancedGraphTable;
+import com.virtuslab.gitmachete.frontend.vfsutils.GitVfsUtils;
+
+@ExtensionMethod({GitVfsUtils.class, GitMacheteBundle.class})
+@CustomLog
+public abstract class BaseTraverseAction extends BaseGitMacheteRepositoryReadyAction implements IBranchNameProvider {
+
+  @Override
+  public LambdaLogger log() {
+    return LOG;
+  }
+
+  public static final String SHOW_TRAVERSE_INFO = "git-machete.traverse.approval.show";
+
+  @Override
+  @UIEffect
+  protected void onUpdate(AnActionEvent anActionEvent) {
```

Comment:
A general sanity check: is it ever possible to run another traverse when a traverse is ongoing? 🤔

Reply:
(ofc it shouldn't, that would be a race-condition-ridden nightmare)

Reply:
Huh in general AFAICS one is able to quickly click e.g. Push or Rebase actions in time windows between the modal dialogs (i.e. when background tasks are running) 😅 although it's at best a tech debt item, not really something to be addressed right now